### ISSUE TITLE
application view (sometimes) shows only blank canvas

### ISSUE THREAD

**Issue by stevesea** [argoproj/argo-cd#1282]
using 0.12-rc4, and chrome, I can sometimes get the application view into a weird state where nothing shows up in the canvas. 

browser console shows this error: 
```
react-dom.production.min.js:198 TypeError: Cannot read property 'x' of undefined
    at application-resource-tree.tsx:20
    at Array.forEach (<anonymous>)
    at pn (application-resource-tree.tsx:19)
    at Zo (react-dom.production.min.js:150)
    at Ni (react-dom.production.min.js:178)
    at Ga (react-dom.production.min.js:232)
    at Ka (react-dom.production.min.js:233)
    at Os (react-dom.production.min.js:249)
    at Cs (react-dom.production.min.js:248)
    at Es (react-dom.production.min.js:245)
ha @ react-dom.production.min.js:198
```
seems like it’s got something to do with filters being applied. It seems to mostly happen when I go to the Application I've created for declarative definition of ArgoCD itself... 

If I get it in that weird state... I can navigate back to a different application, clear the graph view filters, then go back to my 'argocd' application, and the view works.

**Comment by stevesea** [comment]
I was able to reproduce it... 

So, two applications 'argocd' and 'cert-manager'. After clearing the graph view filters, i go to cert-manager, select it to only display serivce accounts in the graph.

I navigate back and to the 'argocd' application, and the whole canvas disappears. I go back (and do F5 refresh), then  to cert-manager, and either select list view or clear the filter, then I can go back to the argocd application and things are ok

**Comment by alexmt** [comment]
Ah, your application manages itself and if filter doesn't include Application kind then app tree removes top graph node and dies. Fixed by https://github.com/argoproj/argo-cd-ui/commit/1b41aba841afa431610abe88091ed1375f50e757

Thank you for reporting it!

**Comment by stevesea** [comment]
confirmed fix with 0.12-rc6
